packages/eslint-plugin/src/rules/no-throw-literal.ts

```diff
@@ -39,7 +43,7 @@ export default util.createRule({
         }
       }
 
-      const baseTypes = type.getBaseTypes() ?? [];
+      const baseTypes = checker.getBaseTypes(type as ts.InterfaceType);
```

Comment:
interesting. Maybe we should add a lint rule against using `type.getBaseTypes()`, considering it fails in generic cases.